sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -176,16 +194,67 @@ internal class EGLCore(
     return EGL10.EGL_SUCCESS
   }
 
-  /**
-   * Checks for EGL errors.
-   */
-  private fun checkEglErrorNoException(msg: String): Boolean {
+  @MainThread
+  internal fun addRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererErrorLock.withLock {
+      rendererSetupErrorListenerSet.add(listener)
+      if (accumulatedRendererErrorList.isNotEmpty()) {
+        accumulatedRendererErrorList.forEach { error ->
+          listener.onError(error)
+        }
+        accumulatedRendererErrorList.clear()
```

Comment:
wonder if we should clear the error list, we could keep them and only clear on destroy the renderer. And document that all the `addRendererStateListener` will return all the egl errors.

Asking because how would user know which `addRendererStateListener` call is made "first" if they add the listener multiple places in their code base? And the listener itself doesn't report this "isFirst" flag back to the caller. Imo we should make this API consistent no matter if it's called the first time or multiple times after the initial subscription.

Reply:
done!